File: SemestrWork/src/main/java/com/itis/servletsapp/Database/FilesService.java
package com.itis.servletsapp.Database;

import java.io.InputStream;
import java.io.OutputStream;

//TODO: дз на 16.10
public interface FilesService {
    FileInfo saveFileToStorage(InputStream file, String originalFileName, String contentType, Long size);
    void readFileFromStorage(Long fileId, OutputStream outputStream);
    FileInfo getFileInfo(Long fileId);
}

File: stalker-shop/src/main/java/com/itis/stalkershop/services/interfaces/FilesService.kt
package com.itis.stalkershop.services.interfaces

import com.itis.stalkershop.models.UploadedFile
import com.itis.stalkershop.models.UploadedFileDto
import java.io.InputStream
import java.io.OutputStream

interface FilesService {
    fun saveFileToStorage(
        file: InputStream,
        originalFileName: String,
        contentType: String,
        size: Long
    ): UploadedFile

    fun writeFileFromStorage(
        fileId: Long,
        outputStream: OutputStream
    )

    fun getFileInfo(
        fileId: Long
    ): UploadedFile
}